scripts/report_scripts/variantreport_p_sample.Rmd

```diff
@@ -525,59 +525,68 @@ Called SNVs:
 # file (and maybe rule)
-# get all possible variants
-all_variants <- colnames (msig_simple[, -which( names(msig_simple) %in% "muts" )])
-output_variants <- file.path(csv_output_dir, "data_variant_plot.csv")
-# 1. write dataframe with this information here
-output_variant_plot <- data.frame (samplename = character(),
+output_variants <- file.path(variants_output_dir, paste0(sample_name, "_variants.csv"))
+output_variant_plot <- data.frame(samplename = character(),
                                    dates = character(),
                                    location_name = character(),
                                    coordinates_lat = character(),
                                    coordinates_long = character())
-# add columns for all possible variants to the dataframe
-for (variant in all_variants) {
-  output_variant_plot[,variant] <- numeric()
-}
-meta_data <- c (samplename = sample_name,
-                dates = date,
-                location_name = location_name,
-                coordinates_lat = coordinates_lat,
-                coordinates_long = coordinates_long)
-
-output_variant_plot <- bind_rows(output_variant_plot, meta_data)
-
-# get rownumber for current sample
-sample_row <- which(grepl(sample_name, output_variant_plot$samplename))
-
-# write mutation frequency values to df
-for (i in all_variants){
-  if (i %in% df$name){
-    # check if variant already has a column
-    if (i %in% colnames(output_variant_plot)) {
-        output_variant_plot[sample_row,][i] <- df$value[df$name == i]
-        output_variant_plot <- output_variant_plot %>% mutate(others = 1-rowSums(across(all_of(all_variants)), na.rm = TRUE))
-}
+if (!executeDeconvolution) {
+  # if no signatur mutation found write empty output file
+  # TODO: sombody should check whether this empty file with header is enough, or a more sensible default is required
+  write.table(output_variant_plot, output_variants,
+    sep = "\t",
+    na = "NA", row.names = FALSE, quote = FALSE
+  )
+} else {
+  # get all possible variants
+  all_variants <- colnames(msig_simple[, -which(names(msig_simple) %in% "muts")])
+  # add columns for all possible variants to the dataframe
+  for (variant in all_variants) {
+    output_variant_plot[,variant] <- numeric()
+  }
+  meta_data <- c (samplename = sample_name,
+                  dates = date,
+                  location_name = location_name,
+                  coordinates_lat = coordinates_lat,
+                  coordinates_long = coordinates_long)
+
+  output_variant_plot <- bind_rows(output_variant_plot, meta_data)
+
+  # get rownumber for current sample
+  sample_row <- which(grepl(sample_name, output_variant_plot$samplename))
+
+  # write mutation frequency values to df
+  for (i in all_variants){
+    if (i %in% df$name){
+      # check if variant already has a column
+      if (i %in% colnames(output_variant_plot)) {
```

Comment:
shouldn't this condition be merged with the above?  This would reduce unnecessary nesting.